lib/oscServer.js

```diff
@@ -1,72 +1,72 @@
-const OSC = require('osc-js')
-const osc = new OSC({ plugin: new OSC.DatagramPlugin() })
-
-const log = ({ address, args }) => {
-  console.dir(address)
-  console.dir(args)
-}
-
-const send = (address, message) => {
-  osc.send(new OSC.Message(address, message))
-}
-
-osc.on('/test', async message => {
-  log(message)
-})
-
-osc.on('/assert', async message => {
-  message.args.forEach(fact => {
-    osc.context.client.assert(fact)
-  })
-  await osc.context.client.flushChanges()
-})
-
-osc.on('/retract', async message => {
-  message.args.forEach(fact => {
-    osc.context.client.retract(fact)
-  })
-  await osc.context.client.flushChanges()
-})
-
-osc.on('/select', async message => {
-  let facts = message.args
-  await osc.context.client.select(facts).doAll(assertions => {
-    send('/assertions', JSON.stringify(assertions))
-  })
-})
+class LivingRoomOscServer {
+  constructor (client) {
+    this.client = client
+    this.connections = new Map()
+    this.messageHandlers = {
+      '/echo': ({connection, args}) => {
+        this.send(connection, '/echo', args)
+      },
+      '/assert': ({args}) => {
+        args.forEach(fact => {
+          this.client.assert(fact)
+        })
+        this.client.flushChanges()
+      },
+      '/retract': ({args}) => {
+        args.forEach(fact => {
+          this.client.retract(fact)
+        })
+        this.client.flushChanges()
+      },
+      '/select': ({connection, args}) => {
+        this.client.select({facts: args})
+          .then(({assertions}) => {
+            this.send(connection, '/assertions', JSON.stringify(assertions))
+          })
+      },
+      '/subscribe': ({connection, args}) => {
+        this.client.subscribe({facts: args})
+          .then(() => {
+            this.send(connection, '/subscriptions', JSON.stringify(args))
+          })
+      }
+    }
+  }
 
-osc.on('open', () => {
-  let queries = null
-  let timeout = null
+  send (connection, address, string) {
+    connection.send({
+      address,
+      args: [ { type: 's', value: string } ]
+    })
+  }
 
-  send('/connection', 'randomid')
+  listen (localPort = 41234, localAddress = '0.0.0.0') {
+    const { UDPPort } = require('osc')
+    const osc = new UDPPort({ localAddress, localPort })
 
-  osc.on('updateSubscription', subscriptions => {
-    queries =
-      typeof subscriptions === 'string' ? [subscriptions] : subscriptions
-  })
+    osc.on(`ready`, () => {
+      const name = process.env.LIVING_ROOM_NAME || require('os').hostname()
```

Comment:
I feel like you push this block to a `publishBonjour` somewhere in `src` and do it that way.